arkouda/segarray.py

```diff
@@ -106,23 +110,47 @@ def __init__(self, segments, values, copy=False, lengths=None, grouping=None):
                 raise ValueError("Segments must start at zero and be less than values.size")
         elif values.size > 0:
             raise ValueError("Cannot have non-empty values with empty segments")
-        if copy:
-            self.segments = segments[:]
-            self.values = values[:]
-        else:
-            self.segments = segments
-            self.values = values
-        self.size = segments.size
-        self.valsize = values.size
+
+        # create the server message
+        rep_msg = generic_msg(
+            cmd="segArr-assemble",
+            args={
+                "segments": segments,
+                "values": values,
+            },
+        )
+
+        try:
+            # parse the return message from building the server object
+            fields = rep_msg.split()
+            self.name = fields[1]
+            self.dtype = fields[2]
+            self.size = int(fields[3])
+            self.ndim = int(fields[4])
+
+            # remove comma from 1 tuple with trailing comma
+            if fields[5][len(fields[5]) - 2] == ",":
+                fields[5] = fields[5].replace(",", "")
+            self.shape = [int(el) for el in fields[5][1:-1].split(",")]
```

Comment:
I'm curious if this `if` hits and the commas are replaced with empty string, are you still expecting there to be something to split on?

Reply:
This logic is to remove a trailing comma, so in the event of `(n,)`. However, in cases of `(n,m)` the if will not hit. This allows the logic to be used in both instances.